homeassistant/components/tplink/switch.py

```diff
@@ -38,9 +39,48 @@ async def async_setup_entry(
     else:
         entities.append(SmartPlugSwitch(device, coordinator))
 
+    entities.append(SmartPlugLedSwitch(device, coordinator))
+
     async_add_entities(entities)
 
 
+class SmartPlugLedSwitch(CoordinatedTPLinkEntity, SwitchEntity):
+    """Representation of switch for the LED of a TPLink Smart Plug."""
+
+    coordinator: TPLinkDataUpdateCoordinator
+
+    def __init__(
+        self, device: SmartDevice, coordinator: TPLinkDataUpdateCoordinator
+    ) -> None:
+        """Initialize the LED switch."""
+        super().__init__(device, coordinator)
+
+        self._attr_entity_category = ENTITY_CATEGORY_CONFIG
+        self._attr_name = f"{device.alias} (LED)"
+        self._attr_unique_id = f"{legacy_device_id(device)}_led"
```

Comment:
```suggestion
        self._attr_unique_id = f"{self.device.mac}_led"
```

We don't need to use the legacy compat here since this is new

Reply:
Does it make sense though to have an ID that might be different from the parent's id ?
Instead of having `<switch id>_led` corresponding to the `<switch id>` entity, we might end-up with `<mac>_<switch id>_led` corresponding to the `<switch id>` entity.

Reply:
We are going to migrate away from the legacy device ids in the future.  It would be nice to not have to migrate this one as well

Reply:
I'll update, then! :)